src/include/catalog/index_schema.h

```diff
@@ -107,6 +106,38 @@ class IndexSchema {
      */
     bool Nullable() const { return static_cast<bool>(packed_type_ & MASK_NULLABLE); }
 
+    /**
+     * Default constructor for deserialization
+     */
+    Column() = default;
+
+    /**
+     * @return column serialized to json
+     */
+    nlohmann::json ToJson() const {
+      nlohmann::json j;
+      j["name"] = name_;
+      j["type"] = Type();
+      j["max_varlen_size"] = MaxVarlenSize();
+      j["nullable"] = Nullable();
+      j["oid"] = oid_;
+      j["definition"] = definition_;
+      return j;
+    }
+
+    /**
+     * Deserializes a column
+     * @param j serialized column
+     */
+    void FromJson(const nlohmann::json &j) {
+      name_ = j.at("name").get<std::string>();
+      SetTypeId(j.at("type").get<type::TypeId>());
+      SetMaxVarlenSize(j.at("max_varlen_size").get<uint16_t>());
+      SetNullable(j.at("nullable").get<bool>());
+      SetOid(j.at("oid").get<indexkeycol_oid_t>());
+      definition_ = j.at("definition").get<std::shared_ptr<parser::AbstractExpression>>();
```

Comment:
This line won't work, use the `parser::expression::DeserializeExpression` method.